mlrun/datastore/redis.py

```diff
@@ -34,27 +35,25 @@ def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
         self.headers = None
 
         self.endpoint = self.endpoint or mlrun.mlconf.redis.url
-        if schema == "ds":
-            datastore_profile = datastore_profile_read(name)
-            if not datastore_profile:
-                raise ValueError(f"Failed to load datastore profile '{name}'")
-            if datastore_profile.type != "redis":
-                raise ValueError(
-                    f"Trying to use profile of type '{datastore_profile.type}' as redis datastore"
-                )
-            self._redis_url = datastore_profile.url_with_credentials()
-            self.secure = datastore_profile.is_secured()
+        parsed_endpoint = urlparse(self.endpoint)
+        if parsed_endpoint.scheme != "redis" and parsed_endpoint.scheme != "rediss":
+            parsed_endpoint = urlparse(f"{schema}://{self.endpoint}")
+        self.secure = parsed_endpoint.scheme == "rediss"
+
+        if parsed_endpoint.username or parsed_endpoint.password:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Provide Redis username and password only via secrets"
+            )
+        user = self._get_secret_or_env("REDIS_USER", "")
+        password = self._get_secret_or_env("REDIS_PASSWORD", "")
+        host = parsed_endpoint.hostname
+        port = parsed_endpoint.port if parsed_endpoint.port else "6379"
```

Comment:
Let's move `6379` to be a config parameter - we want to avoid such numeric constants in the code.